jobs/payment-jobs/tasks/ejv_partner_distribution_task.py

```diff
@@ -53,308 +54,221 @@ def create_ejv_file(cls):
-                                                     PaymentMethod.EFT.value])) \
+        reversals = base_query.filter(InvoiceModel.invoice_status_code.in_([InvoiceStatus.REFUNDED.value,
+                                                                            InvoiceStatus.REFUND_REQUESTED.value,
+                                                                            InvoiceStatus.CREDITED.value])) \
             .filter(InvoiceModel.disbursement_status_code == DisbursementStatus.COMPLETED.value) \
-            .filter(InvoiceModel.corp_type_code == partner.code) \
             .all()
-        current_app.logger.info(invoices)
-        return invoices
+
+        disbursement_rows = []
+        for invoice, payment_line_item, distribution_code in transactions + reversals:
+            disbursement_rows.append(Disbursement(
+                bcreg_distribution_code=distribution_code,
+                partner_distribution_code=distribution_code.partner_distribution_code,
+                line_item=DisbursementLineItem(
+                    amount=payment_line_item.total,
+                    flow_through=f'{invoice.id:<110}',
+                    description_identifier=f'#{invoice.id}',
+                    is_reversal=invoice.invoice_status_code in [InvoiceStatus.REFUNDED.value,
+                                                                InvoiceStatus.REFUND_REQUESTED.value,
+                                                                InvoiceStatus.CREDITED.value],
+                    disbursement_type=EJVLinkType.INVOICE.value,
+                    identifier=invoice.id,
+                    is_legacy=True
+                )
+            ))
+        # ################################################################# END OF Legacy way of handling disbursements.
+
+        # Partner disbursements - New
+        partner_disbursements = db.session.query(PartnerDisbursementsModel,
+                                                 PaymentLineItemModel,
+                                                 DistributionCodeModel) \
+            .join(PaymentLineItemModel, PaymentLineItemModel.invoice_id == PartnerDisbursementsModel.target_id,
+                  PartnerDisbursementsModel.target_type == EJVLinkType.INVOICE.value) \
+            .join(DistributionCodeModel,
+                  DistributionCodeModel.distribution_code_id == PaymentLineItemModel.fee_distribution_id) \
+            .filter(PartnerDisbursementsModel.status_code.is_(None)) \
+            .filter(PartnerDisbursementsModel.corp_type_code == partner.code) \
+            .filter(DistributionCodeModel.stop_ejv.is_(False) | DistributionCodeModel.stop_ejv.is_(None)) \
+            .order_by(DistributionCodeModel.distribution_code_id, PaymentLineItemModel.id) \
+            .all()
+
+        for partner_disbursement, distribution_code in partner_disbursements:
+            suffix = 'PR' if partner_disbursement.disbursement_type == EJVLinkType.PARTIAL_REFUND else 'PD'
```

Comment:
Wanted something different to show up on the flow_through or description_identifier